clang_bind/cmake_frontend.py

```diff
@@ -0,0 +1,162 @@
+import os
+import json
+
+
+class Codemodel:
+    def __init__(self, reply_dir):
+        self.reply_dir = reply_dir
+        self.codemodel = {}
+        self.codemodel_targets = {}
+
+    def set_codemodel(self, codemodel_file=None):
+        if not codemodel_file:
+            for file in os.listdir(self.reply_dir):
+                if file.startswith("codemodel"):
+                    codemodel_file = os.path.join(self.reply_dir, file)
+        with open(codemodel_file) as f:
+            self.codemodel = json.load(f)
+
+    def get_codemodel(self):
+        if not self.codemodel:
+            self.set_codemodel()
+        return self.codemodel
+
+    def set_codemodel_targets(self, codemodel=None):
+        if not codemodel:
+            codemodel = self.get_codemodel()
+        configurations = codemodel.get("configurations", [])
+        for configuration in configurations:
+            targets = configuration.get("targets", [])
+            for target in targets:
+                self.codemodel_targets.update({target["name"]: target["jsonFile"]})
+
+    def get_codemodel_targets(self):
+        if not self.codemodel_targets:
+            self.set_codemodel_targets()
+        return self.codemodel_targets
+
+
+class Target:
+    def __init__(self, reply_dir, target_name):
+        self.reply_dir = reply_dir
+        self.target_file = Codemodel(reply_dir).get_codemodel_targets().get(target_name)
```

Comment:
If this is the expected usage, we don't need a class for Codemodel since there's no state to be managed. Wouldn't just a simple function(s) work as well?
```py
def get_targets(reply_dir):
    # find the codemodel file
    # read it
    # iterate on configurations to get targets
    # send back the targets
```

Reply:
Removed the class taking into account the suggestion. Used a private function in the front-end class.